src/lib/fs/path.c

```diff
@@ -293,3 +341,389 @@ make_path_absolute(char *fname)
   return absfname;
 #endif /* defined(_WIN32) */
 }
+
+/**** Helper functions of type glob_predicate_fn used for glob processing ****/
+
+/** Returns true if <b>file_type</b> represents an existing file (even if
+ * empty). Returns false otherwise. */
+static bool
+is_file(file_status_t file_type, const char *pattern_after)
+{
+  (void)pattern_after;
+  return file_type != FN_ERROR && file_type != FN_NOENT && file_type != FN_DIR;
+}
+
+/** Returns true if <b>file_type</b> represents an existing directory. Returns
+ * false otherwise. */
+static bool
+is_dir(file_status_t file_type, const char *pattern_after)
+{
+  (void)pattern_after;
+  return file_type == FN_DIR;
+}
+
+#ifdef _WIN32
+/** Returns true if the path where <b>file_type</b> was obtained from
+ * concatenated with <b>pattern_after</b> is a complete path and points to an
+ * existing file or directory. Returns false otherwise. */
+static bool
+no_pattern_after(file_status_t file_type, const char *pattern_after)
+{
+  return file_type != FN_ERROR && file_type != FN_NOENT && !*pattern_after;
```

Comment:
This might be clearer and more future-proof as `(is_file() || is_dir()) && ...`.

Reply:
Suggestion applied in ae426f9